src/app/services/baw-api/model.service.spec.ts

```diff
@@ -0,0 +1,2565 @@
+        expect(user).toBeTruthy();
+        done();
+      },
+      () => {
+        expect(true).toBeFalsy("Service should not generate error");
+      }
+    );
+
+    const req = httpMock.expectOne({
+      url: config.getConfig().environment.apiRoot + "/broken_path",
+      method: "PATCH"
+    });
+
+    expect(req.request.body).toEqual({ name: "name" });
+
+    req.flush({
+      meta: {
+        status: 200,
+        message: "OK"
+      },
+      data: {
+        id: 1,
+        name: "name",
+        case_conversion: {
+          test_convert: "test"
+        }
+      }
+    } as APIResponse);
+  });
+
+  it("update should handle complex single value", done => {
+    const service: ModelService<MockModel> = TestBed.get(ModelService);
+    service["update"]("/broken_path", {
+      caseConversion: {
+        testConvert: "test"
+      }
+    }).subscribe(
+      (user: MockModel) => {
+        expect(user).toBeTruthy();
+        done();
+      },
+      () => {
+        expect(true).toBeFalsy("Service should not generate error");
+      }
+    );
+
+    const req = httpMock.expectOne({
+      url: config.getConfig().environment.apiRoot + "/broken_path",
+      method: "PATCH"
+    });
+
+    expect(req.request.body).toEqual({
+      case_conversion: {
+        test_convert: "test"
+      }
+    });
+
+    req.flush({
+      meta: {
+        status: 200,
+        message: "OK"
+      },
+      data: {
+        id: 1,
+        name: "name",
+        case_conversion: {
+          test_convert: "test"
+        }
+      }
+    } as APIResponse);
+  });
+
+  it("update should handle multiple values", done => {
+    const service: ModelService<MockModel> = TestBed.get(ModelService);
+    service["update"]("/broken_path", {
+      name: "name",
+      caseConversion: {
+        testConvert: "test"
+      }
+    }).subscribe(
+      (user: MockModel) => {
+        expect(user).toBeTruthy();
+        done();
+      },
+      () => {
+        expect(true).toBeFalsy("Service should not generate error");
+      }
+    );
+
+    const req = httpMock.expectOne({
+      url: config.getConfig().environment.apiRoot + "/broken_path",
+      method: "PATCH"
+    });
+
+    expect(req.request.body).toEqual({
+      name: "name",
+      case_conversion: {
+        test_convert: "test"
+      }
+    });
+
+    req.flush({
+      meta: {
+        status: 200,
+        message: "OK"
+      },
+      data: {
+        id: 1,
+        name: "name",
+        case_conversion: {
+          test_convert: "test"
+        }
+      }
+    } as APIResponse);
+  });
+
+  it("update should handle error", done => {
+    const service: ModelService<MockModel> = TestBed.get(ModelService);
+    service["update"]("/broken_path", null).subscribe(
+      () => {
+        expect(true).toBeFalsy("Service should not generate data response");
+      },
+      (err: APIErrorDetails) => {
+        expect(err).toBeTruthy();
+        expect(err).toEqual({
+          stat
```

Comment:
I've managed to understand a little of this file, but:

- it is way too long
- has way too much repetition in it

Can you find some ways to break it up?

Reply:
I have two immediate methods I could use to split up the unit tests. Either I create multiple `describe` functions inside the file, or I can split the file into multiple smaller files.

The first method will maintain a large file, however I can group unit tests by the method it tests, and the unit test output will separate the tests into different groups similar to separate unit test files. 

The second method of splitting the file up would have a similar effect to the first method in terms of output from the unit testing framework, however it would add unnecessary files to the directory. I would likely split the files by method name resulting in the following file name format: `model.<method>.spec.ts`.

Which method do you prefer.

As for the repetition, some of the no filter vs filter tests could be removed with regards to error handling (as that is independent of filter), however this would make the assumption that someone wont break that independence in the future which I'm not a huge fan of. While I understand there is a lot of repetition, I'm unsure how I would reduce it without reducing the coverage of the testing.

Reply:
I'll let you decide.

If you can factor out enough common code, then keep it one file.

Normally I'd baulk at more files, but 2.5K lines in a file are unwieldy. Breaking out into methods is fine if you can't factor out helper methods/common code etc